mantis-publish/mantis-publish-core/src/test/java/io/mantisrx/publish/internal/discovery/mantisapi/DefaultMantisApiClientTest.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2022 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.publish.internal.discovery.mantisapi;
+
+import com.netflix.archaius.DefaultPropertyFactory;
+import com.netflix.archaius.api.PropertyRepository;
+import com.netflix.archaius.config.DefaultSettableConfig;
+import com.netflix.mantis.discovery.proto.AppJobClustersMap;
+import com.netflix.mantis.discovery.proto.JobDiscoveryInfo;
+import com.netflix.spectator.api.DefaultRegistry;
+import com.netflix.spectator.ipc.http.HttpClient;
+import io.mantisrx.publish.config.SampleArchaiusMrePublishConfiguration;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+public class DefaultMantisApiClientTest {
+    @Disabled
+    @Test
+    public void manualTest() throws Exception {
+        final String apiServerHostName = "your test host name here";
+        final String jobClusterName = "your test job cluster name here";
+        final DefaultRegistry registry = new DefaultRegistry();
+        final Properties props = new Properties();
+        final DefaultSettableConfig config = new DefaultSettableConfig();
+
+        props.put("mantis.publish.discovery.api.hostname", apiServerHostName);
+        config.setProperties(props);
+
+        final PropertyRepository propsRepo = DefaultPropertyFactory.from(config);
+        final DefaultMantisApiClient defaultMantisApiClient = new DefaultMantisApiClient(new SampleArchaiusMrePublishConfiguration(propsRepo), HttpClient.create(registry));
+        final CompletableFuture<JobDiscoveryInfo> jobDiscoveryInfoCompletableFuture = defaultMantisApiClient.jobDiscoveryInfo(jobClusterName);
+        final JobDiscoveryInfo jobDiscoveryInfo = jobDiscoveryInfoCompletableFuture.get(1, TimeUnit.SECONDS);
+
+        System.out.println(jobDiscoveryInfo);
```

Comment:
Could you please test key-value pairs in `assertEquals`, `assertTrue`, etc instead? I don't think it's a good idea to just print to console for testing.

Reply:
@hmitnflx I'm not sure what to check because those values will depend on the job cluster you point the client at. If you have some pointers to setup an api server + source job in a unit test, I can try that and add some asserts then.